src/sonarr.ts

```diff
@@ -1,136 +1,151 @@
 import { sonarrApi } from './fetch.ts';
 import { cleanTitle, hasExclusionTag } from './utils.ts';
 
+type SonarrApiClient = NonNullable<typeof sonarrApi>;
+
+interface SonarrSeries {
+  id: number;
+  title: string;
+  tvdbId: number;
+  tags: number[];
+}
+
+interface SonarrEpisode {
+  id: number;
+  seriesId: number;
+  tvdbId: number;
+  seasonNumber: number;
+  episodeNumber: number;
+  monitored: boolean;
+}
+
+interface SonarrTag {
+  id: number;
+  label: string;
+}
+
+async function getSonarrSeriesList(
+  api: SonarrApiClient,
```

Comment:
Instead of having to do this, let's make a couple of changes:

Let's remove the `SONARR_API_KEY` check here so the client is always defined:
https://github.com/Shraymonks/unmonitorr/blob/09469df922552df67329ab25c44539dddfe7a390/src/fetch.ts#L20

And instead add the checks here and only unmonitor if it's defined:
https://github.com/Shraymonks/unmonitorr/blob/09469df922552df67329ab25c44539dddfe7a390/src/jellyfin.ts#L26-L30
https://github.com/Shraymonks/unmonitorr/blob/09469df922552df67329ab25c44539dddfe7a390/src/plex.ts#L51-L56


Then we can just directly use `sonarApi` throughout here